### DISCUSSION TITLE
Request header and response header signature verification

### DISCUSSION THREAD

**Discussion by hulinux** [directus/directus#10179] [Feature Requests]
I want to add signature information to the request header and response header in each request and response, such as:

signature：9E4498DECAF8461BB00747C1E2B37F1E

Only after verifying the signature on the server side will it respond to the request or verify the signature on the receiving side before the data is considered legal. In the current version, I don’t see which extension can be implemented.

Is there or is it possible to design ‘request:before’ and ‘response:before’ events to implement this mechanism

This signature is used to verify data, which is different from "access token"

**Comment by rijkvanzanten** [comment]
Heya! Do you have some more background information on how this functionality is supposed to work? Sounds like something that might be interesting to add to Directus as a core functionality rather than an extension. 

> Is there or is it possible to design ‘request:before’ and ‘response:before’ events to implement this mechanism

Yeah, I can see the usefulness in having a filter hook for those types 🙂

**Reply by hulinux** [reply]
In some cases, the requesting side and the service side have agreed on a certain verification mechanism (for example: verifying the data using the encryption method agreed by both parties), then the request data needs to be verified before the request is sent, and the verification result information is put in the request header , Use the same verification method to verify the request data on the server side, compare the verification data in the request header to verify that both parties have verified and are consistent (can be regarded as a custom encrypted communication method)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
